transient/config.py

```diff
@@ -0,0 +1,261 @@
+       during deserialization
+    """
+
+    config = fields.Str(allow_none=True)
+    copy_in_before = fields.List(fields.Str(), allow_none=True)
+    copy_out_after = fields.List(fields.Str(), allow_none=True)
+    copy_timeout = fields.Int(allow_none=True)
+    image = fields.List(fields.Str(), allow_none=True)
+    image_backend = fields.Str(allow_none=True)
+    image_frontend = fields.Str(allow_none=True)
+    name = fields.Str(allow_none=True)
+    prepare_only = fields.Bool(allow_none=True)
+    qemu_args = fields.List(fields.Str(), allow_none=True)
+    qmp_timeout = fields.Int(allow_none=True)
+    shutdown_timeout = fields.Int(allow_none=True)
+    ssh_command = fields.Str(allow_none=True)
+    ssh_bin_name = fields.Str(allow_none=True)
+    ssh_port = fields.Int(allow_none=True)
+    ssh_timeout = fields.Int(allow_none=True)
+    ssh_user = fields.Str(allow_none=True)
+    ssh_console = fields.Bool(allow_none=True)
+    ssh_with_serial = fields.Bool(allow_none=True)
+    shared_folder = fields.List(fields.Str(), allow_none=True)
+
+    # marshmallow's decorator post_load() is untyped, forcing create_args()
+    # to be untyped. Therefore, we ignore it to silence the type checker
+    @post_load  # type: ignore
+    def create_config(self, data: Dict[Any, Any], **kwargs: Dict[Any, Any]) -> Config:
+        """Returns the Config dictionary after a schema is loaded and validated
+        """
+        return Config(**data)
+
+
+def replace_hyphens_with_underscores_in_dict_keys(
+    dictionary: Dict[Any, Any]
+) -> Dict[Any, Any]:
+    """Replaces hyphens in the dictionary keys with underscores
+
+       This is the expected key format for TransientConfigSchema
+    """
+    final_dict = {}
+    for k, v in dictionary.items():
+        # Perform this method recursively for sub-directories
+        if isinstance(dictionary[k], dict):
+            value = replace_hyphens_with_underscores_in_dict_keys(v)
+            final_dict[k.replace("-", "_")] = value
+        else:
+            final_dict[k.replace("-", "_")] = v
+
+    return final_dict
+
+
+def parse_config_file(config_file_path: str) -> MutableMapping[str, Any]:
+    """Parses the given config file and returns the contents as a dictionary
+    """
+    with open(config_file_path) as file:
+        config_file = file.read()
+
+    try:
+        parsed_config_file = toml.loads(config_file)
+    except RuntimeError as error:
+        print(error, file=sys.stderr)
+        sys.exit(1)
+
+    return parsed_config_file
+
+
+def get_line_number_of_string_in_config_file(string: str, config_file_path: str) -> int:
+    """Returns the line number where the config file contains the given string
+    """
+    with open(config_file_path) as config_file:
+        for line_number, line in enumerate(config_file, start=1):
+            if string in line:
+                return line_number
+
+    print("Line not found", file=sys.stderr)
```

Comment:
Definitely shouldn't print this cryptic message here.

Reply:
Print statement deleted. Technically that print statement should have been moved to the calling function and invoked if if the line number is -1.

Technically this shouldn't happen because the matching string comes directly from the config file itself, but I wasn't sure what to do if the line number is -1 for whatever magical reason.